internal/pkg/aws/elbv2/elbv2.go

```diff
@@ -153,3 +156,62 @@ func (t *TargetHealth) HealthStatus() *HealthStatus {
 func (t *TargetHealth) targetID() string {
 	return aws.StringValue(t.Target.Id)
 }
+
+// LoadBalancer contains information about a given load balancer.
+type LoadBalancer struct {
+	ARN            string
+	Name           string
+	DNSName        string
+	Scheme         string // "internet-facing" or "internal"
+	SecurityGroups []string
+}
+
+// LoadBalancer returns select information about a load balancer.
+func (e *ELBV2) LoadBalancer(id string) (*LoadBalancer, error) {
+	var input *elbv2.DescribeLoadBalancersInput
+	if arn.IsARN(id) {
+		input = &elbv2.DescribeLoadBalancersInput{
+			LoadBalancerArns: []*string{aws.String(id)},
+		}
+	} else {
+		input = &elbv2.DescribeLoadBalancersInput{
+			Names: []*string{aws.String(id)},
+		}
+	}
+	output, err := e.client.DescribeLoadBalancers(input)
+	if err != nil {
+		return nil, fmt.Errorf("describe load balancer %q: %w", id, err)
+	}
+	lb := output.LoadBalancers[0]
+	return &LoadBalancer{
+		ARN:            aws.StringValue(lb.LoadBalancerArn),
+		Name:           aws.StringValue(lb.LoadBalancerName),
+		DNSName:        aws.StringValue(lb.DNSName),
+		Scheme:         aws.StringValue(lb.Scheme),
+		SecurityGroups: aws.StringValueSlice(lb.SecurityGroups),
+	}, nil
+}
+
+// Listener contains information about a listener.
+type Listener struct {
+	ARN      *string
```

Comment:
Just a dumb question: Is there a reason why we will only have `ARN`, `Port` and `Protocol`. Why not `SSL certificate` and `SecurityPolicy` in case of HTTPS protocol? Is this  because customer need to bring certificates and security policies on their own.

Reply:
Good question. Yes. And we can add those fields later if we need to.